source/slang/slang-state-serialize.cpp

```diff
@@ -1072,10 +1074,234 @@ struct LoadContext
+            cmd.addArg(Profile(src.profile).getName());
+
+            if (src.targetFlags & SLANG_TARGET_FLAG_PARAMETER_BLOCKS_USE_REGISTER_SPACES)
+            {
+                cmd.addArg("-parameter-blocks-use-register-spaces");
+            }
+
+            switch (src.floatingPointMode)
+            {
+                case FloatingPointMode::Fast:
+                {
+                    cmd.addArg("-fp-mode");
+                    cmd.addArg("fast");
+                    break;
+                }
+                case FloatingPointMode::Precise:
+                {
+                    cmd.addArg("-fp-mode");
+                    cmd.addArg("precise");
+                    break;
+                }
+                default: break;
+            }
+
+#if 0
+            // If there is output state (like output filenames) add here
+            if (src.outputStates.getCount())
+            {
+                RefPtr<EndToEndCompileRequest::TargetInfo> dstTargetInfo(new EndToEndCompileRequest::TargetInfo);
+                request->targetInfos[dstTarget] = dstTargetInfo;
+
+                for (const auto& srcOutputStateOffset : src.outputStates)
+                {
+                    const auto& srcOutputState = base.asRaw(srcOutputStateOffset);
+
+                    SLANG_ASSERT(srcOutputState.entryPointIndex < requestState->entryPoints.getCount());
+
+                    String entryPointPath;
+                    if (srcOutputState.outputPath)
+                    {
+                        entryPointPath = base.asRaw(srcOutputState.outputPath)->getSlice();
+                    }
+
+                    dstTargetInfo->entryPointOutputPaths.Add(srcOutputState.entryPointIndex, entryPointPath);
+                }
+            }
+#endif
+        }
+    }
+
+    {
+        const auto& srcPaths = requestState->searchPaths;
+        for (Index i = 0; i < srcPaths.getCount(); ++i)
+        {
+            cmd.addArg("-I");
+            cmd.addArg(base.asRaw(base.asRaw(srcPaths[i]))->getSlice());
+        }
+    }
+
+    {
+        for (const auto& define : requestState->preprocessorDefinitions)
+        {
+            StringBuilder builder;
+            builder << "-D" << base.asRaw(base.asRaw(define).first)->getSlice();
+            if (base.asRaw(define).second)
+            {
+                builder << "=" << base.asRaw(base.asRaw(define).second)->getSlice();
+            }
+        }
+    }
+
+    {
+        const auto& srcTranslationUnits = requestState->translationUnits;
+
+        for (Index i = 0; i < srcTranslationUnits.getCount(); ++i)
+        {
+            const auto& srcTranslationUnit = base.asRaw(srcTranslationUnits[i]);
+
+            //context.loadDefines(srcTranslationUnit.preprocessorDefinitions, dstTranslationUnit->preprocessorDefinitions);
+
+#if 0
+            if (srcTranslationUnit.moduleName)
+            {
+                moduleName = base[srcTranslationUnit].moduleName->getSlice());
+            }
+#endif
```

Comment:
I would expect this module name on a translation unit to translate over to `-module-name <whatever>` eventually, once we implement the flag to actually apply per-translation-unit.